rebar.config.sample

```diff
@@ -42,6 +42,14 @@
 %% yecc files to compile first
 {yrl_first_files, []}.
 
+%% == Application Discovery ==
+
+%% Extensions for discovering application resource files. Rebar3 discovers
+%% applications by searching for these files. When building the application
+%% information, rebar will use the first discovered files in the list.
+{application_resource_extensions, [
+    "app.src.script", "app.src", "mix.exs"
+]}
```

Comment:
The code seems to have an expectation of the extension's final fragment, i.e. `should end in .app, .src, .script, or .exs (for plugins adapting mix code)` and have the comment mention the intent, with something like `intended to let profiles compose with special app files by prefixing extensions` so people know whether they need or want this.